typeclasses/characters.py

```diff
@@ -91,6 +75,71 @@ def str(self, amount):
 
         self.db.slots = {}
 
+    # helper method, checks if stat is valid
+    def find_stat(self, stat):
+        if stat in self.db.primary_traits:
+            return self.db.primary_traits[stat]
+        elif stat in self.db.secondary_traits:
+            return self.db.secondary_traits[stat]
+        else:
+            return None
+
+    # helper method, checks if race gets extra language points
+    def determine_language_points(self):
+        if self.bonuses['languages']:
+            return self.bonuses['languages']
+
+    def base_stat(self, stat, amount):
+        # sets the secondary traits
+        if stat == 'vitality':
+            base_stat('health', amount)
+            base_stat('stamina', amount)
+            base_stat('fortitude', amount)
+            base_stat('reflex', amount)
+
+        if stat == 'intelligence':
+            bonus_language_points = determine_language_points()
+            base_stat('languages', amount + bonus_language_points)
+            base_stat('will', amount)
+              
+        # as per the OA blue rulebook mana can never exceed 10 (page 49)
+        if stat = 'mana':
+            if amount > 10:
+                self.db.secondary_traits['mana'].base = 10
+                return
+            else:
+                self.db.secondary_traits['mana'].base = amount
+                return
+
+        stat_valid = find_stat(stat)
+        if stat_valid:
+            stat_valid.base = amount
+        else:
+            return False
+
+    def mod_stat(self, stat, amount):
+        # as per the OA blue rulebook mana can never exceed 10 (page 49)
+        if stat = 'mana':
+            if (self.db.secondary_traits['mana'].base + amount) > 10:
+                self.db.secondary_traits['mana'].mod = 10
```

Comment:
Keep in mind `current` will return `base + mod`, so if `base + amount > 10`, then setting `mod` to 10...what if base is already 10? `current` will now be 20. This logic is not correct and will have to be thought through carefully.